### PR TITLE
Remove redundant prompts and inappropriate retreat options. Bump version and fix old version processing

### PR DESCRIPTION
**Pull request by simon33-2**
Hmm, having a small amount of trouble with git. Once a commit has been pushed I'm not sure how to remove it from a given branch, so I killed "RemovePrompts2"

Previous attempts are #1599 & #1615.

Basically, this resolves most or all of #1270. Also removes a redundant prompt for submerging subs against only air units.

This PR and #1583 break serialisation in certain cases so move the version number on and establish a new file for the latest version. If there is a better way of doing this then I am all ears.

The major reason why isn't it achievable to download the actual source file which contains the version number is because that needs to be merged in two phases. The build will fail if code which refers to the source is merged at the same time as the source being updated, if you follow me. This is a one time problem though. If this is merged, I reckon just update EngineVersionProperties to refer to GameEnginePropertyFileReader. Then the version is stored in a single place and a jar file knows what version it is, if not which build.

### CODE REVIEW

File: src/main/java/games/strategy/triplea/delegate/MustFightBattle.java [hunk 1]
```diff
@@ -815,9 +833,24 @@ private void pushFightLoopOnStack(final boolean firstRun) {
 
       @Override
       public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+        if( Match.someMatch( m_attackingUnits, Matches.unitHasAttackValueOfAtLeast(1) )
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MustFightBattle.java]: This is a Checkstyle whitespace violation (just run the Eclipse formatter to fix it).

File: latest_version_new.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+LATEST=1.9.1.0
```
Comment (DanVanAtta) [inline latest_version_new.properties]: What's the background on why we need a `latest_version_new.properties`?

**Comment by DanVanAtta** [comment]
@simon33-2  some blockers:
- multiple changes in same PR. It's going to be difficult to get everything discussed and tested.
- It's very significant to upgrade game compatibility version and it is impactful to the  players. This update can potentially be blocked until we have enough changes queued to justify a version upgrade. (One reason why eliminating compatibility issues is so important).
- The version number problems/handling should be discussed a good bit more. I'm not quite sold on adding back in a new place where we read game engine version number and am not clear on the intent of the new property file.

 @simon33-2 given the above, what are your thoughts and intent on how to move forward?

**Comment by simon33-2** [comment]
We need to go to 1.9.1 anyway. We already have at least one change in the master code line which can break serialisation in some cases. Besides, this improvement is worth breaking backwards compatibility.

Re: latest_version file.
We cannot update the current file at present because that causes all existing instances of 1.9.0 up to the currently released (not just built) ones to hang.

My idea of going to a new file would be that would allow 1.9.1 users to be notified when that version is deprecated. However, there is a better idea which is possible; I don't know how this was handled in 1.8 although it does appear that the jar file did indeed know what version it was. By incorporating the version in the source we can completely remove the latest_version* in later versions. If you are worrying about this, I think one option is to just have a 1.9.1 think that the latest version is 1.9.0 in the first build and then quickly update it before deploying to triplea-game.org or marking it as the latest release. Would that make you happy about this aspect? Keen to hear your thoughts.

Re: GameEnginerProperty.ENGINE_VERSION
I don't think it is a bug since you should be using the constant however I will take your point that it is more brittle than it should be.

I'll push up a fix for this one.

**Comment by DanVanAtta** [comment]
I see, I'm not that opposed to going to 1.9.1 

We are in a tough spot though since the intended update mechanism is broken. My hope/intent was we could work out what was needed to ensure we could do that upgrade. The game is supposed to notify you when there is a new version available and prompt you to download it. otherwise the engine does know which version it is by parsing the game_engine.properties map.

The new file is an interesting idea. We will fail to notify users of the new version. Since that is broken anyways, i can understand the work around. I think if we are explicit enough with some lobby messaging and let enough time go by wiht a patched version available for release, that those who are 'stuck' will be few, and they will hopefully re-download.

**Comment by simon33-2** [comment]
So which way are you leaning:
1) New file
2) Non functional latest_version check
3) something else?

**Comment by DanVanAtta** [comment]
New file would have the problem that it would be difficult to migrate off of it. 

I would lean towards two, non functional, but that assumes a good bit of time to organically get people on a patched code base. On the other hand, migrating to a new file, and planning to stay on that file would probably be the best transition possible.

**Comment by simon33-2** [comment]
I'm inclined to agree, given that I'm suggesting a single point of truth solution. I'll push that up shortly.

**Pull request closed without merging**